internal/certgen/k8s.go

```diff
@@ -14,51 +14,42 @@
 package certgen
 
 import (
-	"path"
-
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func newTLSSecret(secretname, namespace string, keyPEM, certPEM []byte) *corev1.Secret {
-
+func newTLSSecret(name string, namespace string, data map[string][]byte) *corev1.Secret {
 	return &corev1.Secret{
 		Type: corev1.SecretTypeTLS,
 		TypeMeta: metav1.TypeMeta{
 			Kind:       "Secret",
 			APIVersion: "v1",
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      secretname,
+			Name:      name,
 			Namespace: namespace,
 			Labels: map[string]string{
 				"app": "contour",
 			},
 		},
-		Data: map[string][]byte{
-			corev1.TLSCertKey:       certPEM,
-			corev1.TLSPrivateKeyKey: keyPEM,
-		},
+		Data: data,
 	}
 }
 
-func newCertOnlySecret(secretname, namespace, certfilename string, certPEM []byte) *corev1.Secret {
-
+func newCertOnlySecret(name string, namespace string, data map[string][]byte) *corev1.Secret {
```

Comment:
Now that both `newTLSSecret()` and `newCertOnlySecret()` both take map of data as parameter, the only difference left seems to be the secret `Type`.  Maybe consider combining these into single function and also pass secret type as a parameter?

Reply:
Fixed.